worker/jobs/bulkUpsertSamplesJob.js

```diff
@@ -56,17 +56,16 @@ module.exports = (job, done) => {
           return Promise.resolve();
         } else {
           successCount++;
+          if (featureToggles.isFeatureEnabled('publishSampleInPromiseChain')) {
+            // Wait for publish to complete before resolving the promise.
```

Comment:
Technically this only waits for the db request to complete, the actual publish is still done asynchronously. I think it would be good to be clear about this in the comments.

Approving since this is a minor concern.